diff --git a/src/builtin.c b/src/builtin.c
index 0715a332..1b6ab734 100644
--- a/src/builtin.c
+++ b/src/builtin.c
@@ -940,6 +940,26 @@ builtin `%s' requested by frozen file is not supported"), arg));
 
 /* Exit code from last "syscmd" command.  */
 static int sysval;
+#ifdef __MVS__
+ #if (__CHARSET_LIB == 1)
+#   include <stdio.h>
+#   include <stdlib.h>
+
+    static int setccsid(int fd, int ccsid) 
+    {
+      attrib_t attr;
+      int rc;
+
+      memset(&attr, 0, sizeof(attr));
+      attr.att_filetagchg = 1;
+      attr.att_filetag.ft_ccsid = ccsid;
+      attr.att_filetag.ft_txtflag = 1;
+
+      rc = __fchattr(fd, &attr, sizeof(attr));
+      return rc;
+    }
+  #endif
+#endif
 
 static void
 m4_syscmd (struct obstack *obs MAYBE_UNUSED, int argc, token_data **argv)
@@ -1034,6 +1054,11 @@ m4_esyscmd (struct obstack *obs, int argc, token_data **argv)
       close (fd);
       return;
     }
+#ifdef __MVS__
+ #if (__CHARSET_LIB == 1)
+   setccsid(fileno(pin), 819);
+ #endif
+#endif
   while (1)
     {
       size_t avail = obstack_room (obs);
@@ -1435,6 +1460,11 @@ mkstemp_helper (struct obstack *obs, const char *me, const char *pattern,
     }
   else
     {
+#ifdef __MVS__
+ #if (__CHARSET_LIB == 1)
+      setccsid(fd, 819);
+ #endif
+#endif
       close (fd);
       /* Remove NUL, then finish quote.  */
       obstack_blank_fast (obs, -1);
