***************
*** 981,984 ****
--- 981,997 ----
  }
  
+ #ifdef __MVS__
+ static int __chgfdccsid(int fd, unsigned short ccsid) {
+   attrib_t attr;
+   memset(&attr, 0, sizeof(attr));
+   attr.att_filetagchg = 1;
+   attr.att_filetag.ft_ccsid = ccsid;
+   if (ccsid != FT_BINARY) {
+     attr.att_filetag.ft_txtflag = 1;
+   }
+   return __fchattr(fd, &attr, sizeof(attr));
+ }
+ #endif
+ 
  static void
  m4_esyscmd (struct obstack *obs, int argc, token_data **argv)
***************
*** 1023,1026 ****
--- 1036,1043 ----
  #else
    pin = fdopen (fd, "r");
+ #endif
+ 
+ #ifdef __MVS__
+   __chgfdccsid(fileno(pin), 819);
  #endif
  
